src/foam/blob/RestBlobService.java

```
@@ -0,0 +1,164 @@
+      long chunks = (long) Math.ceil((double) size / (double) BUFFER_SIZE);
+      Buffer buffer = new Buffer(BUFFER_SIZE, ByteBuffer.allocate(BUFFER_SIZE));
+      os = connection.getOutputStream();
+
+      while ( chunk < chunks ) {
+        buffer = blob.read(buffer, chunkOffset(chunk));
+        byte[] buf = buffer.getData().array();
+        os.write(buf, 0, (int) buffer.getLength());
+        buffer.getData().clear();
+        chunk++;
+      }
+
+      os.flush();
+
+      if ( connection.getResponseCode() == HttpURLConnection.HTTP_OK ) {
+        is = connection.getInputStream();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+        String json = "";
+        String line = null;
+        while ( (line = reader.readLine()) != null ) {
+          json += line;
+        }
+
+        result = (IdentifiedBlob) (new foam.lib.json.JSONParser()).parseString(json, IdentifiedBlob.class);
+        result.setX(getX());
+      } else {
+        throw new RuntimeException("upload fail");
+      }
+    } catch ( MalformedURLException e ) {
+      e.printStackTrace();
+    } catch ( IOException e ) {
+      e.printStackTrace();
+    } finally {
+      closeSource(is, os);
+      return result;
+    }
+  }
+
+  @Override
+  public Blob find_(X x, Object id) {
+    InputStream is = null;
+    HttpURLConnection connection = null;
+    Blob blob = null;
+    try {
+      URL url = new URL(this.address_ + "/" + id.toString());
+      connection = (HttpURLConnection) url.openConnection();
+
+      connection.setRequestMethod("GET");
+      connection.connect();
+
+      if ( connection.getResponseCode() == HttpURLConnection.HTTP_OK ) {
+        is = connection.getInputStream();
+        blob = new InputStreamBlob(is);
+      } else {
+        throw new RuntimeException("download fail");
+      }
+    } catch ( MalformedURLException e ) {
+      e.printStackTrace();
+    } catch ( IOException e ) {
+      e.printStackTrace();
```

Again might want to rethrow this exception.